src/ui/Share/Dialog.svelte

```diff
@@ -0,0 +1,201 @@
+<script context="module" lang="ts">
+  import { dialogs } from '@/ui/Dialog'
+  import ShareDialog from './Dialog.svelte'
+
+  export enum SHARE_TYPE {
+    Watchlist = 'watchlist',
+    Alert = 'alert',
+    Chart = 'chart',
+  }
+
+  const SECRET_LINK_TAG = '__SECRET_LINK_TAG__'
+  const SECRET_TEXT_TAG = '__SECRET_TEXT_TAG__'
+
+  export const MEDIA_LINKS = [
+    {
+      link: `https://twitter.com/home?status=${SECRET_TEXT_TAG}%0Alink%3A%20${SECRET_LINK_TAG}`,
+      title: 'Twitter',
+      background: '#56A8DC',
+    },
+    {
+      link: `https://santiment.net/discord`,
+      title: 'Discord',
+      background: '#5B65EA',
+    },
+    {
+      link: `https://reddit.com/submit?title=${SECRET_TEXT_TAG}&url=${SECRET_LINK_TAG}`,
+      title: 'Reddit',
+      background: '#EB5528',
+    },
+    {
+      link: `https://telegram.me/share/url?text=${SECRET_TEXT_TAG}&url=${SECRET_LINK_TAG}`,
+      title: 'Telegram',
+      background: '#5AAADB',
+    },
+  ]
+
+  export const showShareDialog = (props) =>
+    dialogs.show(ShareDialog, Object.assign({ strict: true }, props))
+</script>
+
+<script lang="ts">
+  import Svg from '@/ui/Svg/svelte'
+  import Toggle from '@/ui/Toggle.svelte'
+  import Dialog from '@/ui/Dialog'
+  import { copy } from '@/utils'
+
+  export let shareType
+  export let shareLink
+  export let shareText
+  export let onChangePrivacy = () => {}
```

Comment:
1. `export let onChangePrivacy = noop` (from `@/utils`
2. new line between props and state